src/Compilers/CSharp/Test/Semantic/Semantics/PropertyFieldKeywordTests.cs

```diff
@@ -1350,9 +1378,394 @@ public S(int arg)
+            comp.VerifyDiagnostics(
+                // error CS1007: Property accessor already defined
+                Diagnostic(ErrorCode.ERR_DuplicateAccessor, "get").WithLocation(6, diagnosticColumn)
+                );
+
+            var tree = comp.SyntaxTrees[0];
+            var token = tree.GetRoot().DescendantTokens().Single(t => t.IsKind(SyntaxKind.NumericLiteralToken) && (int)t.Value == numericLiteralToSpeculate);
+
+            var model = comp.GetSemanticModel(tree);
+            var identifier = SyntaxFactory.ParseExpression("field");
+            model.TryGetSpeculativeSemanticModel(token.SpanStart, (IdentifierNameSyntax)identifier, out var speculativeModel);
+
+            Assert.Empty(comp.GetTypeByMetadataName("C").GetFieldsToEmit());
+
+            var fieldKeywordSymbolInfo = speculativeModel.GetSymbolInfo(identifier);
+            Assert.Null(fieldKeywordSymbolInfo.Symbol);
+            Assert.Equal(0, accessorBindingData.NumberOfPerformedAccessorBinding);
+            Assert.Empty(comp.GetTypeByMetadataName("C").GetFieldsToEmit());
+        }
+
+        [Theory]
+        [InlineData("get => 0; get => 1;", 0)]
+        [InlineData("get => 0; get => 1;", 1)]
+        [InlineData("get { return 0; } get => 1;", 0)]
+        [InlineData("get { return 0; } get => 1;", 1)]
+        [InlineData("get => 0; get { return 1; }", 0)]
+        [InlineData("get => 0; get { return 1; }", 1)]
+        [InlineData("get { return 0; } get { return 1; }", 0)]
+        [InlineData("get { return 0; } get { return 1; }", 1)]
+        public void SpeculativeSemanticModel_FieldInDuplicateAccessorWhereFirstAccessorDoesNotUseField_BindSpeculatedFirst(string accessors, int numericLiteralToSpeculate)
+        {
+            var comp = CreateCompilation($@"
+class C
+{{
+    public double P
+    {{
+        {accessors}
+    }}
+}}
+");
+            var accessorBindingData = new SourcePropertySymbolBase.AccessorBindingData();
+            comp.TestOnlyCompilationData = accessorBindingData;
+
+            var tree = comp.SyntaxTrees[0];
+            var token = tree.GetRoot().DescendantTokens().Single(t => t.IsKind(SyntaxKind.NumericLiteralToken) && (int)t.Value == numericLiteralToSpeculate);
+
+            var model = comp.GetSemanticModel(tree);
+            var identifier = SyntaxFactory.ParseExpression("field");
+            model.TryGetSpeculativeSemanticModel(token.SpanStart, (IdentifierNameSyntax)identifier, out var speculativeModel);
+            var fieldKeywordSymbolInfo = speculativeModel.GetSymbolInfo(identifier);
+
+            Assert.Equal("System.Double C.<P>k__BackingField", comp.GetTypeByMetadataName("C").GetFieldsToEmit().Single().ToTestDisplayString());
+            Assert.Equal("System.Double C.<P>k__BackingField", fieldKeywordSymbolInfo.Symbol.ToTestDisplayString(includeNonNullable: true));
             Assert.Equal(0, accessorBindingData.NumberOfPerformedAccessorBinding);
         }
     }
```

Comment:
>} [](http://example.com/codeflow?start=4&length=1)

I think it will be useful to add similar speculation tests replacing body of accessors that don't use `field` keyword with bodies that use "field" identifier. For example, replace what used to be
```
{
    return 0;
}
```

with 
```
{
    in field = 0;
    return field;
}
```

Reply:
I am still tracking this thread.

Reply:
@AlekseyTs Sorry for missing this comment earlier. I added some compiler tests for this. For expression evaluator, I don't know how can I specify at which point the evaluation happens? For example:

```csharp
    if (GetBoolValue())
    {
        int field = 10;
        return field; // evaluating here should use the local
     }

    return field; // Evaluating here should use the backing field.
```

Reply:
> For expression evaluator, I don't know how can I specify at which point the evaluation happens? 

The `ExpressionCompilerTests.Evaluate()` helper method has an `atLineNumber:` parameter. There are several tests that use that, typically by including a `#line` directive in the source.

Reply:
Thank you @cston for guidance! I added the tests.